### DISCUSSION TITLE
Cop for disallowing TODO comments

### DISCUSSION THREAD

**Discussion by hi-tech-jazz** [rubocop/rubocop#10597] [Ideas]
Would a good idea be to disallow `TODO`, `FIXME`, and `OPTIMIZE`comments?

Depending on the size and workflow of a specific project, the use of such comments may be considered bad practice. Instead, the unit of work associated with _changing some code in the near or distant future_ is better handled as an issue on the Github repo, a Trello card, or a ticket in whatever project management tool the project is using.

**Comment by koic** [comment]
Currently (RuboCop 1.28.2) there is no such cop against. I'm not sure whether it's acceptable because I've never met such the practice, but it's technically implementable. So If the cop will be provided, it is disabled by default.

**Comment by pirj** [comment]
Annotations don't prevent you from filing tickets.

One benefit of keeping annotations in the code is that references in tickets rot if you refactor your code, while annotations migrate along with them. You can refer the ticket from an annotation, too.
